sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,650 @@
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> deleteTag(String tag) {
+        return this.deleteTagWithResponse(tag)
+            .flatMap((Response<Void> res) -> Mono.empty());
+    }
+
+    /**
+     * Get repository properties.
+     *
+     * @return repository properties.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<RepositoryProperties>> getPropertiesWithResponse() {
+        return withContext(context -> this.getPropertiesWithResponse(context));
+    }
+
+    Mono<Response<RepositoryProperties>> getPropertiesWithResponse(Context context) {
+        try {
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
+            return this.serviceClient.getPropertiesWithResponseAsync(repositoryName, context)
+                .map(res -> Utils.mapResponse(res, Utils::mapRepositoryProperties))
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Get repository properties.
+     *
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return repository properties.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<RepositoryProperties> getProperties() {
+        return this.getPropertiesWithResponse().map(Response::getValue);
+    }
+
+    /**
+     * Get registry artifact properties.
+     *
+     * @param tagOrDigest tag or digest associated with the blob.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return registry artifact properties.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<RegistryArtifactProperties>> getRegistryArtifactPropertiesWithResponse(String tagOrDigest) {
+        return withContext(context -> this.getRegistryArtifactPropertiesWithResponse(tagOrDigest, context));
+    }
+
+    Mono<Response<RegistryArtifactProperties>> getRegistryArtifactPropertiesWithResponse(String tagOrDigest, Context context) {
+        try {
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
+
+            Mono<String> getTagMono = Mono.defer(
+                () -> tagOrDigest.contains(":")
```

Comment:
Should document that this will make a second network call if the tagOrDigest contains `:`. Additionally, this doesn't need to be put into a defer as all network calls are implicitly deferred until subscribed to